flattentool/input.py

```diff
@@ -502,10 +587,12 @@ def fancy_unflatten(self, with_cell_source_map, with_heading_source_map):
         return result, ordered_cell_source_map, heading_source_map
 
 
-def extract_list_to_error_path(path, input):
+def extract_list_to_error_path(path, input, index=None):
```

Comment:
I think this function is a bit confusing. The actual implementation is fine once I got my head around it, so maybe it just requires a bit of explanation.